src/uucore/src/lib/parser/parse_size.rs

```diff
@@ -175,6 +181,59 @@ impl<'parser> Parser<'parser> {
             .ok_or_else(|| ParseSizeError::size_too_big(size))
     }
 
+    /// Explicit u128 alias for `parse()`
+    pub fn parse_u128(&self, size: &str) -> Result<u128, ParseSizeError> {
+        self.parse(size)
+    }
+
+    /// Same as `parse()` but tries to return u64
+    pub fn parse_u64(&self, size: &str) -> Result<u64, ParseSizeError> {
+        match self.parse(size) {
+            Ok(num_u128) => {
+                let num_u64 = match u64::try_from(num_u128) {
+                    Ok(n) => n,
+                    Err(_) => return Err(ParseSizeError::size_too_big(size)),
+                };
+                Ok(num_u64)
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Same as `parse_u64()`, except returns `u64::MAX` on overflow
+    /// GNU lib/coreutils include similar functionality
+    /// and GNU test suite checks this behavior for some utils
+    pub fn parse_u64_max(&self, size: &str) -> Result<u64, ParseSizeError> {
+        let result = self.parse_u64(size);
+        match result {
+            Ok(_) => result,
+            Err(error) => {
+                if let ParseSizeError::SizeTooBig(_) = error {
+                    Ok(u64::MAX)
+                } else {
+                    Err(error)
+                }
+            }
+        }
+    }
+
+    /// Same as `parse_u128()`, except returns `u128::MAX` on overflow
+    /// /// GNU lib/coreutils include similar functionality
+    /// and GNU test suite checks this behavior for some utils
```

Comment:
Which ones do this? This feels like something unrelated to adding `R` and `Q` and I'd like to consider it separately, because it can also happen for `Y` for example.

Reply:
specifically `split` does this and their lib includes function in support of that (i.e. return long int max on overflow if I recall correctly). This comment is actually from code that was already there for u64 variant - just duplicated it for u128 version of that